src/components/Searchbar.js

```diff
@@ -0,0 +1,182 @@
+import React, { useState } from 'react';
+import { Form, Button, Row, Col } from 'react-bootstrap';
+import './Search.css'
+
+const Searchbar = (props) => {
+    const [state, setState] = useState({
+      description: '',
+      python: false,
+      java : false,
+      c : false,
+      game: false,
+      webapp: false,
+      mobileapp: false 
+    });
+  
+    const handleInputChange = (event) => {
+      const { name, value } = event.target;
+      if (name === 'python') {
+        setState((prevState) => ({ ...state, [name]: !prevState.python }));
+      } else {
+        setState({ ...state, [name]: value });
+      }
+    };
+
+    const handleInputChange2 = (event) => {
+        const { name, value } = event.target;
+        if (name === 'java') {
+          setState((prevState) => ({ ...state, [name]: !prevState.java }));
+        } else {
+          setState({ ...state, [name]: value });
+        }
+      };
+
+      const handleInputChange3 = (event) => {
+        const { name, value } = event.target;
+        if (name === 'c') {
+          setState((prevState) => ({ ...state, [name]: !prevState.c }));
+        } else {
+          setState({ ...state, [name]: value });
+        }
+      };
+
+      const handleInputChange4 = (event) => {
+        const { name, value } = event.target;
+        if (name === 'game') {
+          setState((prevState) => ({ ...state, [name]: !prevState.game }));
+        } else {
+          setState({ ...state, [name]: value });
+        }
+      };
+
+      const handleInputChange5 = (event) => {
+        const { name, value } = event.target;
+        if (name === 'webapp') {
+          setState((prevState) => ({ ...state, [name]: !prevState.webapp}));
+        } else {
+          setState({ ...state, [name]: value });
+        }
+      };
+
+      const handleInputChange6 = (event) => {
+        const { name, value } = event.target;
+        if (name === 'mobileapp') {
+          setState((prevState) => ({ ...state, [name]: !prevState.mobileapp}));
+        } else {
+          setState({ ...state, [name]: value });
+        }
+      };
+  
+    const handleSearch = (event) => {
+      event.preventDefault();
+      console.log(state);
+    };
+  
+    return (
+      <container id="main-container" className="d-grid h-100">
+      
+        <div className="search">
+        <Form className="search-form" onSubmit={handleSearch}>
+          <div className = "search-section">
+          <Row>
+            <Col>
+              <Form.Group controlId="description">
+                <Form.Control
+                  type="text"
+                  name="description"
+                  value={state.description || ''}
```

Comment:
Could be `value = {state.description}` here and `state.description` can be initialized to `''`.